hazelcast/src/main/java/com/hazelcast/internal/nearcache/impl/store/AbstractNearCacheRecordStore.java

```diff
@@ -314,7 +314,8 @@ public void put(K key, V value) {
         R oldRecord = null;
         try {
             record = valueToRecord(value);
-            onRecordCreate(key, record);
+            onRecordCreate(record);
+            initInvalidationMetaData(record, key, keyData);
```

Comment:
`initInvalidationMetaData` should be done at reservation phase, doing it in publishing phase can break eventual consistency.